com.spotify.Client.appdata.xml

```diff
@@ -1,12 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
-  <id>com.spotify.Client.desktop</id>
+  <id>com.spotify.Client</id>
```

Comment:
I guess you should add `<provides><id>com.spotify.Client.desktop</id></provides>` along with the rename.

Reply:
Why?

Reply:
Everything I read as far as flatpak and appstream says that the id should be the same as the flatpak/dbus id. I looked around and the convention seems to be to use .desktop. But that is not correct according to both sets of docs.

Nesting id in provides also seems incorrect. According to the docs:

"The provides tag and its children describe the public interfaces this application provides. A public interface can be anything which other applications, which are not part of the upstream project, can access or reference. This includes binaries and libraries. Private interfaces should never be added to a provides tag."

Reply:
So software knows the two different IDs are equal.

Reply:
That doesn't make any sense to me. A nested tag doesn't sound like an interface? Does anyone else do that? Is anything that's going to read it understand what that means?

Reply:
All stores read it, I'm not sure what you are confused by.

`<provides><id>` is documented: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent

Reply:
"<id/>
Contains the component-ID of another software component. The presence of this tag indicates that the software component containing it is able to provide all functionality of the one referenced in the <provides/> ↪ <id/> tag.
This is useful in case a component-id had to be renamed in the past, e.g. because its domain-name changed."

Yep my bad found it.

Sorry will do.

Reply:
Done